akka-stream/src/main/scala/akka/stream/impl/JsonObjectParser.scala

```diff
@@ -36,6 +37,469 @@ import scala.annotation.switch
+
+    }
+
+    object InitialState extends ParserState {
+      override def toString: String = "Initial"
+
+      @tailrec
+      final override def proceed(input: Byte, pp: JsonObjectParser): Unit = {
+        if (input == SquareBraceStart) {
+          pp.skip()
+          pp.enterState(MainArrayBeforeElement)
+        } else if (isWhitespace(input)) {
+          pp.skip()
+          if (pp.keepSeeking) {
+            proceed(pp.nextInput, pp)
+          }
+        } else if (input == DoubleQuote) {
+          pp.take()
+          pp.enterState(InOuterString(NotArrayAfterElement, pp))
+        } else if (input == CurlyBraceStart) {
+          pp.take()
+          pp.enterContainer(InOuterObject, NotArrayAfterElement)
+        } else if (!isWhitespace(input)) {
+          pp.take()
+          pp.enterState(InOuterNaked(NotArrayAfterElement, pp))
+        } else {
+          throw new FramingException(s"Invalid JSON encountered at position [$pp.pos] of [$pp.buffer]")
+        }
+      }
+    }
+
+    /**
+     * We are in this state whenever we're inside a JSON Array-style stream, before any element
+     */
+    private object MainArrayBeforeElement extends ParserState {
+      override def toString: String = "MainArrayBeforeElement"
+
+      @tailrec
+      final override def proceed(input: Byte, pp: JsonObjectParser): Unit = input match {
+        case _ if isWhitespace(input) ⇒
+          pp.skip()
+          if (pp.keepSeeking) {
+            proceed(pp.nextInput, pp)
+          }
+        case Comma ⇒
+          throw new FramingException(s"Invalid JSON encountered at position [${pp.pos}] of [${pp.buffer}]")
+        case SquareBraceEnd ⇒
+          pp.skip()
+          pp.enterState(AfterMainArray)
+        case SquareBraceStart ⇒
+          pp.take()
+          pp.enterContainer(InOuterArray, MainArrayAfterElement)
+        case DoubleQuote ⇒
+          pp.take()
+          pp.enterState(InOuterString(MainArrayAfterElement, pp))
+        case CurlyBraceStart ⇒
+          pp.take()
+          pp.enterContainer(InOuterObject, MainArrayAfterElement)
+        case _ if !isWhitespace(input) ⇒
+          pp.take()
+          pp.enterState(InOuterNaked(MainArrayAfterElement, pp))
+        case _ ⇒
+          throw new FramingException(s"Invalid JSON encountered at position [${pp.pos}] of [${pp.buffer}]")
+      }
+
+    }
+
+    private object AfterMainArray extends ParserState {
+      override def toString: String = "AfterMainArray"
+
+      @tailrec
+      final override def proceed(input: Byte, pp: JsonObjectParser): Unit = (input: @switch) match {
+        case _ if isWhitespace(input) ⇒
+          pp.skip()
+          if (pp.keepSeeking) {
+            proceed(pp.nextInput, pp)
+          }
+        case _ ⇒
+          throw new FramingException(s"Invalid JSON encountered at position [${pp.pos}] of [${pp.buffer}] after closed JSON-style array")
+      }
```

Comment:
Just do an `if-else` instead of pat match since there is just two branches on a boolean condition

Reply:
Doing this. 

Note that this came from using a kind of "cookie-cutter" approach to managing the conditions, and might be the remnants of additional conditions at this specific place. 
While the bytecode is not strictly identical, there should be few hits in practice anyway (whitespace after the end of an array-formatted stream)